File: netapp_api_certificate_authentication/ontap_cert.pl
#!/usr/bin/perl -w

use lib 'NMSDK';
use NaServer;

$cert = "devops_cert.pem";
$key = "devops_cert.key";
$cluster = "svlngen4-c01-trad-gen001";

$server = new NaServer($cluster, 1, 30);
$server->set_transport_type('HTTPS');
$server->set_port('443');
$server->set_style('CERTIFICATE');
$server->set_server_cert_verification(0);
$server->set_client_cert_and_key($cert, $key);

$output = $server->invoke('system-get-version');
if ($output->results_status() eq "failed") {
	die "Error: ", $output->results_reason();
}

$version = $output->child_get_string('version');
print "V: $version\n";

File: netapp_api_certificate_authentication/ontap_cert.py
#!/usr/bin/env python

import sys
sys.path.append("NMSDKpy")
from NaServer import *

cluster = "svlngen4-c01-trad-gen001"
transport = "HTTPS"
port = 443 
style = "CERTIFICATE"
cert = "devops_cert.pem"
key = "devops_cert.key"

s = NaServer(cluster, 1, 30)
s.set_transport_type(transport)
s.set_port(port)
s.set_style(style)
s.set_server_cert_verification(0)
s.set_client_cert_and_key(cert, key)

api = NaElement("system-get-version")
output = s.invoke_elem(api)
if (output.results_status() == "failed"):
    r = output.results_reason()
    print("Failed: " + str(r))
    sys.exit(2)

ontap_version = output.child_get_string("version")
print ("V: " + ontap_version)
